lib/src/foundation/bloc.dart

```diff
@@ -28,27 +28,64 @@ class Bloc {
     return bloc;
   }
 
-  Iterable<String> _steps;
-  int _activeStepIndex;
-
   // The different streams send the featureId that must display/complete.
 
   final StreamController<String> _dismissController =
       StreamController.broadcast();
+
   Stream<String> get outDismiss => _dismissController.stream;
+
   Sink<String> get _inDismiss => _dismissController.sink;
 
   final StreamController<String> _completeController =
       StreamController.broadcast();
+
   Stream<String> get outComplete => _completeController.stream;
+
   Sink<String> get _inComplete => _completeController.sink;
 
   final StreamController<String> _startController =
       StreamController.broadcast();
+
   Stream<String> get outStart => _startController.stream;
+
   Sink<String> get _inStart => _startController.sink;
 
-  String get activeStepId => _steps?.elementAt(_activeStepIndex);
+  /// The steps consist of the feature ids of the features to be discovered.
+  Iterable<String> _steps;
+
+  int _activeStepIndex;
+
+  String get activeFeatureId =>
+      _activeStepIndex == null ? null : _steps?.elementAt(_activeStepIndex);
+
+  /// This is used to determine if the active feature is already shown by
+  /// another [DescribedFeatureOverlay] as [DescribedFeatureOverlay.allowShowingDuplicate]
+  /// requires this.
+  int _activeOverlays;
+
+  int get activeOverlays => _activeOverlays;
+
+  /// [DescribedFeatureOverlay]s use either `activeOverlays++` to add themselves
+  /// to the active overlays or `activeOverlays--` to remove themselves.
+  set activeOverlays(activeOverlays) {
+    // Callers (DescribedFeatureOverlay's) can only either add or remove
+    // themselves, i.e. the difference will always be 1.
+    assert((_activeOverlays - activeOverlays).abs() == 1);
+
+    _activeOverlays = activeOverlays;
+
+    // There is the possibility that two DescribedFeatureOverlays with the same
+    // feature id were active when the feature id was added to _inStart initially.
+    // However, any of them could have had allowShowingDuplicate set to false,
+    // in which case that overlay would not have been opened.
+    //
+    // If the overlay that was showing the active step was disposed now,
+    // which is the only possible cause for activeOverlays == 0 in the setter,
+    // then we can possibly show that other overlay again because it is not
```

Comment:
If you are wondering if this actually works: I wrote a test for this called "Show overlay after duplicate has been removed".